### PR TITLE
Create more filter endpoints

### PR DESCRIPTION
**Pull request by patriciaahuang**
## Overview

The previous search filter included the parameters departure date/time, start location, end location, and the location radius we want to search in. The search only worked if we included all parameters, but the new recent rides homepage requires searching of only the departure date/time, or only the start location, etc. To accommodate this, I expanded on the previous search filter by allowing searching of only a subset of parameters.

## Changes Made

I added 6 new endpoints that allows searching by 
1. only departure date/time
2. only start location
3. only end location
4. departure date/time and start location
5. departure date/time and end location
6. start and end location

I changed the search rides controller to account for the possibility that some parameters may not exist.

## Test Coverage

I tested each of the new endpoints using Postman by creating rides that meet the criteria of the endpoint, and ensured it returned only the rides that matched the specified parameters.

### CODE REVIEW

**Review by joycrwu — APPROVED** [review]
looks good! these are technically all a part of the same url endpoint right

File: src/ride/controllers/search_ride_controller.py [hunk 1]
```diff
@@ -23,70 +23,220 @@ def process(self):
...
-        if response.status_code == 200:
-            end_coords = (
-                response.json()["result"]["geometry"]["location"]["lat"],
-                response.json()["result"]["geometry"]["location"]["lng"],
+
+            # Sort results based on time proximity
+            all_rides = sorted(
+                all_rides,
+                key=lambda ride: (
+                    abs(departure_datetime_object - ride.departure_datetime),
```
Comment (maxwell1231) [inline src/ride/controllers/search_ride_controller.py:47]: Maybe just sort earliest to latest? Not sure, could ask Lindsey!

File: src/ride/views.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ class SearchView(MultipleFieldLookupMixin, generics.RetrieveAPIView):
     serializer_class = RideSerializer
     lookup_fields = ['time', 'start', 'end', 'radius']
 
-    def get(self, request, time, start, end, radius):
+    def get(self, request, time=None, start=None, end=None, radius=0):
```
Comment (maxwell1231) [inline src/ride/views.py]: can we do positive radius (maybe 1) just so it isn't too strict

**Review by maxwell1231 — APPROVED** [review]
Looks good to me, just review comments!

**Pull request merged**